File: python/gn_uuid.py
#!/usr/bin/env python

import uuid
import sys
scientific_name = sys.argv[1]
gn_namespace = uuid.UUID("{90181196-fecf-5082-a4c1-411d4f314cda}")
gn_uuid = uuid.uuid5(gn_namespace, scientific_name)
print("Scientific name: %s" % scientific_name)
print("GlobalNames UUID: %s" %  gn_uuid)

File: go/gn_uuid.go
package main

import (
  "fmt"
  "os"
  "github.com/satori/go.uuid"
)

func main() {
  scientifi_name := os.Args[1]
  gn_name_space := uuid.NewV5(uuid.NamespaceDNS, "globalnames.org")
  gn_uuid := uuid.NewV5(gn_name_space, scientifi_name)
  fmt.Println("Scientific name: " + scientifi_name)
  fmt.Printf("GlobalNames UUID: %s\n", gn_uuid)
}
